packages/ui/src/pages/Continue/IdentifierProfileForm/index.tsx

```diff
@@ -0,0 +1,112 @@
+import { useTranslation } from 'react-i18next';
+
+import Button from '@/components/Button';
+import ErrorMessage from '@/components/ErrorMessage';
+import type { IdentifierInputType } from '@/components/InputFields';
+import { SmartInputField } from '@/components/InputFields';
+import { getGeneralIdentifierErrorMessage, validateIdentifierField } from '@/utils/form';
+
+import * as styles from './index.module.scss';
+
+type Props = {
+  className?: string;
+  // eslint-disable-next-line react/boolean-prop-naming
+  autoFocus?: boolean;
+  defaultType: IdentifierInputType;
+  enabledTypes: IdentifierInputType[];
+
+  onSubmit?: (identifier: IdentifierInputType, value: string) => Promise<void> | void;
+  errorMessage?: string;
+  clearErrorMessage?: () => void;
+};
+
+type FormState = {
+  identifier: string;
+};
+
+const IdentifierProfileForm = ({
+  className,
+  autoFocus,
+  defaultType,
+  enabledTypes,
+  onSubmit,
+  errorMessage,
+  clearErrorMessage,
+}: Props) => {
+  const { t } = useTranslation();
+  const [inputType, setInputType] = useState<IdentifierInputType>(defaultType);
+
+  const {
+    setValue,
+    handleSubmit,
+    control,
+    formState: { errors, isSubmitted },
+  } = useForm<FormState>({
+    reValidateMode: 'onChange',
+    defaultValues: { identifier: '' },
+  });
+
+  const onSubmitHandler = useCallback(
+    async (event?: React.FormEvent<HTMLFormElement>) => {
+      clearErrorMessage?.();
+
+      void handleSubmit(async ({ identifier }, event) => {
+        event?.preventDefault();
+
+        await onSubmit?.(inputType, identifier);
+      })(event);
+    },
+    [clearErrorMessage, handleSubmit, inputType, onSubmit]
+  );
+
+  return (
+    <form className={classNames(styles.form, className)} onSubmit={onSubmitHandler}>
+      <Controller
+        control={control}
+        name="identifier"
+        rules={{
+          required: getGeneralIdentifierErrorMessage(enabledTypes, 'required'),
+          validate: (value) => {
+            const errorMessage = validateIdentifierField(inputType, value);
+
+            if (errorMessage) {
+              return typeof errorMessage === 'string'
+                ? t(`error.${errorMessage}`)
+                : t(`error.${errorMessage.code}`, errorMessage.data);
+            }
+
+            return true;
+          },
+        }}
+        render={({ field }) => (
+          <SmartInputField
+            autoComplete="new-identifier"
+            autoFocus={autoFocus}
+            className={styles.inputField}
+            {...field}
+            currentType={inputType}
+            isDanger={!!errors.identifier}
+            errorMessage={errors.identifier?.message}
+            enabledTypes={enabledTypes}
+            onTypeChange={setInputType}
+            /* Overwrite default input onChange handler  */
+            onChange={(value) => {
+              setValue('identifier', value, { shouldValidate: isSubmitted, shouldDirty: true });
+            }}
```

Comment:
NIT:
The doc says we should avoid manually invoking [setValue](https://react-hook-form.com/api/useform/setvalue) or other APIs related to field updates, but I cannot figure out a proper solution.
There should be no problem, if we encounter inexplicable problems in the future, remember to take a look here.
<img width="942" alt="image" src="https://user-images.githubusercontent.com/10806653/218890468-4f36c5c5-c98f-482f-a20b-e02643517bf2.png">
https://react-hook-form.com/api/usecontroller/controller

Reply:
```ts
const identifier = watch('identifier');

useEffect(() => {
  setIsDirtyOrSomething()
}, [identifier]);
```

Reply:
The tricky thing is we need to merge the country-code selector value and phone-number input field value manually within the InputField component before calling that onChange callbacks.  

The component redefines the onChange callback props, passing in a manipulated value instead of the original input field onChange event handlers. 

@xiaoyijun @wangsijie let me know if there are any better solutions for this use case.  cc: @charIeszhao